[scorecard] Fix commit for models.yaml

driazati opened the pull request:
Instead of always fetching the latest (which makes replicating results
difficult and updates to models.yaml painful), this uses a specific hash
(the current HEAD as of this PR)

YuchenJin commented:
Thanks @LeshengJin for the first Relax pr! This indeed helps a lot with debugging.

Could you also add print support for `ShapeExpr` as well? Printing out symbolic and integer shapes on the python side make sthe debugging easier as well. To do that, you can add a visitor `RelayTextPrinter::VisitExpr_(const ShapeExprNode* op` to `RelayTextPrinter`: https://github.com/tlc-pack/relax/blob/relax/src/printer/text_printer.h#L164.

LeshengJin commented:
Sure, no problem. Thanks for your comment. @YuchenJin

YuchenJin commented on python/tvm/ir/type.py:
```diff
@@ -170,6 +170,10 @@ class TupleType(Type):
     def __init__(self, fields, span: Span = None):
         self.__init_handle_by_constructor__(_ffi_api.TupleType, fields, span)
 
+    def __str__(self):
```
Let's use the unified printer `RelayTextPrinter`, the python side printing will be redirected. We can add a `RelayTextPrinter::VisitType_(const DynTensorTypeNode* node)`, and override this type visitor for TupleType: https://github.com/tlc-pack/relax/blob/relax/src/printer/text_printer.h#L193.

LeshengJin replied:
> Thanks @LeshengJin for the first Relax pr! This indeed helps a lot with debugging. :)
> 
> Could you also add print support for `ShapeExpr` as well? Printing out symbolic and integer shapes on the python side makes the debugging easier as well. To do that, you can add a visitor `RelayTextPrinter::VisitExpr_(const ShapeExprNode* op` to `RelayTextPrinter`: https://github.com/tlc-pack/relax/blob/relax/src/printer/text_printer.h#L164.
> 
> Correction: Since `ShapeExpr` and `DynTensorType` are Relax expr/type, we actually need to use `RelaxScriptPrinter` to print it, so I think we just need to add logics to `PrintFinal` instead of touching `RelayTextPrinter`:
> 
> https://github.com/tlc-pack/relax/blob/3d181851855cfb7fde3e0af90b43ebad9e488d90/src/printer/text_printer.h#L583-L587

If we use `RelaxScriptPrinter` to print `ShapeExpr`, it works well, e.g. `relax.ShapeExpr([tir.IntImm("int64", 10)])` -> `(10,)`. But for `DynTensorType`, the output is in the regular format, e.g., `Tensor[(_, _), "float16"]`. 

Which one is better, `Tensor[(_, _), "float16"]` or `Tensor[rank=2, dtype="float16"]`?

YuchenJin replied:
I definitely prefer `Tensor[rank=2, dtype="float16"]`. :)

yongwww commented on src/printer/relay_text_printer.cc:
```diff
@@ -776,6 +776,12 @@ Doc RelayTextPrinter::VisitType_(const TypeDataNode* node) {
+  doc << "Tensor[rank=" << node->rank << ", dtype=\"" << PrintDType(node->dtype) << "\"]";
+  return doc;
+}
+
```
it would be good if you can add a test case in test_printer.py/test_parser.py

LeshengJin replied:
@yongwww Sure! Thanks!

LeshengJin commented:
Testcases updated. Ready for review! Thanks! @YuchenJin @yongwww

YuchenJin reviewed (approved):
LGTM, thanks @LeshengJin for the contribution! Will merge the PR after Yong's rebase.

yongwww commented:
it was merged, thanks @LeshengJin!!! thanks folks for the review. @ZihengJiang @YuchenJin

driazati committed:
[scorecard] Fix commit for models.yaml

Instead of always fetching the latest (which makes replicating results
difficult and updates to models.yaml painful), this uses a specific hash
(the current HEAD as of this PR)
